File: backend/src/main/java/br/com/visitsafe/dto/release/ServiceProviderReleaseJustificationDTO.java
package br.com.visitsafe.dto.release;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import java.time.OffsetDateTime;

@Data
public class ServiceProviderReleaseJustificationDTO {

    @NotNull
    @Schema(description = "Justification for late exit", required = true)
    private String justification;

    @Schema(description = "Base64 encoded image data")
    private String imageData;

    @Schema(description = "Image MIME type (e.g., image/jpeg, image/png)")
    private String imageType;

    @Schema(description = "Time when the resident was notified about the late exit")
    private String notificationTime;
} 

File: frontend/src/api/models/ServiceProviderReleaseJustificationDTO.ts
export type ServiceProviderReleaseJustificationDTO = {
  /**
   * @description Justification for late exit
   * @type string
   */
  justification: string
  /**
   * @description Base64 encoded image data
   * @type string | undefined
   */
  imageData?: string
  /**
   * @description Image MIME type (e.g., image/jpeg, image/png)
   * @type string | undefined
   */
  imageType?: string
  /**
   * @description Time when the resident was notified about the late exit
   * @type string | undefined
   */
  notificationTime?: string
}